src/nvim/shada.c

```diff
@@ -2714,6 +2723,10 @@ static ShaDaWriteResult shada_write(FileDescriptor *const sd_writer,
   map_foreach_value(&wms->file_marks, val, {
     xfree(val);
   })
+  const char *stored_key = NULL;
+  map_foreach_key(&wms->file_marks, stored_key, {
+    xfree((char *)stored_key);
```

Comment:
how is this different than the `map_foreach_value(&wms->file_marks, val` loop just above this?

Reply:
oh i did not see that `map_foreach_value` and `map_foreach_key` are different names.